basyx/aas/adapter/http.py

```diff
@@ -268,6 +285,9 @@ def json_list(cls, data: Union[str, bytes], expect_type: Type[T], stripped: bool
                 constructor = decoder._construct_asset_information  # type: ignore[assignment]
             elif expect_type is model.SpecificAssetId:
                 constructor = decoder._construct_specific_asset_id  # type: ignore[assignment]
+            elif expect_type is model.Reference:
+                constructor = decoder._construct_reference   # type: ignore[assignment]
+                args.append(model.Submodel)
```

Comment:
I don't think `_construct_reference` accepts any additional positional arguments, so `args.append(model.Submodel)` will probably have to be removed here.